luatest/cbuilder.lua

```diff
@@ -0,0 +1,231 @@
+-- Configuration builder.
+--
+-- Basic usage:
+--
+-- local config = cbuilder_new()
+--     :add_instance('instance-001', {
+--         database = {
+--             mode = 'rw',
+--         },
+--     })
+--     :add_instance('instance-002', {})
+--     :add_instance('instance-003', {})
+--     :config()
+--
+-- By default, all instances are added to replicaset-001 in group-001,
+-- but it's possible to select a different replicaset and/or group:
+--
+-- local config = cbuilder_new()
+--     :use_group('group-001')
+--     :use_replicaset('replicaset-001')
+--     :add_instance(<...>)
+--     :add_instance(<...>)
+--     :add_instance(<...>)
+--
+--     :use_group('group-002')
+--     :use_replicaset('replicaset-002')
+--     :add_instance(<...>)
+--     :add_instance(<...>)
+--     :add_instance(<...>)
+--
+--     :config()
+--
+-- The default credentials and iproto options are added to
+-- setup replication and to allow a test to connect to the
+-- instances.
+--
+-- There is a few other methods:
+--
+-- * :set_replicaset_option('foo.bar', value)
+-- * :set_instance_option('instance-001', 'foo.bar', value)
+--
+-- All the methods supports chaining (returns the builder object
+-- back).
+
+local fun = require('fun')
+
+local log = require('luatest.log')
+
+local has_cluster_config, cluster_config = pcall(require, 'internal.config.cluster_config')
+if not has_cluster_config then
+    log.error([[Declarative configuration works on Tarantool 3.0.0+.
```

Comment:
Shouldn't we explicitly break here (not just to log) if Tarantool is lower than 3.0.0?